src/signalrclient/constants.h

```diff
@@ -5,4 +5,4 @@
 
 #define SIGNALR_VERSION "0.1.0-alpha0"
 #define USER_AGENT "SignalR.Client.Cpp/" SIGNALR_VERSION
-#define MAX_NEGOTIATE_REDIRECTS 100
+#define MAX_NEGOTIATE_REDIRECTS 3
```

Comment:
Another example of our threading story not being present. This will stackoverflow currently if it's higher.

Reply:
This is because these are now callbacks?

Reply:
Yeah, in the "near" future we'll probably talk about what we'll do for threading so some of these callbacks will be dispatched.

Reply:
Still though, 3 redirects seems like it should be handled. How large are these callbacks on the stack? Anyway to push some of these to the heap?

Reply:
This would also only affect synchronous/inline negotiate completions, right? It doesn't seem likely for there to be 100 in a row inline completions to negotiate in practice, but I understand guarding against this anyway.